lib/Myriad/RPC/Message.pm

```diff
@@ -111,8 +111,8 @@ method encode {
             response   => encode_json_text($response),
             trace      => encode_json_text($trace),
         });
-    } catch {
-        Myriad::Exception::RPC::InvalidRequest->new("Bad message encoding")->throw;
+    } catch ($e) {
+        Myriad::Exception::RPC::InvalidRequest->throw(reason => $e);
```

Comment:
same here, also why we used `throw` here instead of `new`?

Reply:
it's the same thing `throw` will create a new exception see here: https://github.com/binary-com/perl-Myriad/blob/master/lib/Myriad/Exception.pm#L38